predicates/traffic/rand_test.go

```diff
@@ -0,0 +1,18 @@
+package traffic_test
+
+import (
+	"math/rand/v2"
+	"sync"
+)
+
+var (
+	testRandMu sync.Mutex
+	testRand   = rand.New(rand.NewPCG(0x5EED_1, 0x5EED_2))
+)
+
+// testRandFloat64 is a helper function to generate fixed sequence of random float64 values for testing.
+func testRandFloat64() float64 {
+	testRandMu.Lock()
+	defer testRandMu.Unlock()
+	return testRand.Float64()
+}
```

Comment:
This is not reusable by multiple tests. Its better to introduce an object to generate random values from the same seed.

Reply:
Fixed